### ISSUE TITLE
2d Galsim interpolants in JAX

### ISSUE THREAD

**Issue by ismael-mendoza** [GalSim-developers/JAX-GalSim#54]
@beckermr implemented a jax version of 2D galsim interpolants in this notebook: https://github.com/beckermr/misc/blob/main/work/jax_dmdet_hacking/interp.ipynb

Might be good idea to integrate them into jax-galsim

**Comment by jecampagne** [comment]
Would be nice to get the residuals in a use-case for the 2D interpolants

**Comment by beckermr** [comment]
Oh yeah! It is on my list. I think though I'll just start a PR with tests to codify what they are explicitly.

**Comment by jecampagne** [comment]
FYI. Along the line to see if your code could be proposed to JAX, I was looking to Scipy 2D interpolation API, but (may be you know) the doc of
[scipy.interpolate.interp2d](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.interp2d.html)
 mention that the API is deprecated since the 1.10.0 version and will be removed in the 1.13.0. The new API is documented 
[here as scipy.interpolate.RegularGridInterpolator](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.RegularGridInterpolator.html#scipy-interpolate-regulargridinterpolator).


Now, if you are interested, I had opened a discussion in May 22 (!) on JAX [here](https://github.com/google/jax/discussions/10689) and I have proposed at that time code that you will find at the end of the thread. It would be nice to make some comparisons if you do not mind. My pb was the edges as usual...
